content/en/serverless/distributed_tracing/serverless_trace_propagation.md

```diff
@@ -0,0 +1,216 @@
+---
+title: Serverless Trace Propagation
+kind: documentation
+further_reading:
+- link: "/tracing/"
+  tag: "Documentation"
+  text: "Explore Datadog APM"
+- link: "/tracing/trace_search_and_analytics/#live-search-for-15-minutes"
+  tag: "Documentation"
+  text: "Live Search"
+---
+
+
+{{< img src="serverless/lambda-non-http-trace.png" alt="Serverless Distributed Non-HTTP Trace"  style="width:100%;" >}}
+
+## Required Setup
+
+Additional instrumentation is sometimes required to see a single, connected trace in Node and Python serverless applications asynchronously triggering Lambda functions. If you are just getting started with monitoring serverless applications in Datadog, first [follow our main installation steps][1] and [read this page on choosing your tracing library][2]. Once you are sending traces from your Lambda functions to Datadog using the [Datadog Lambda Library][3], you may want to follow these steps to connect traces between two Lambda functions in cases such as:
+- Asynchronously triggering Lambda functions via SNS, Kinesis or EventBridge
+- Invoking Lambda functions via non-HTTP protocols such as MQTT
+
+To successfully connect trace context between resources sending traces, you need to:
+- Include Datadog trace context in outgoing events. The outgoing event can originate from a host or Lambda function with `dd-trace` installed.
+- Extract the trace context in the consumer Lambda function.
+
+## Passing Trace Context to Outgoing Events
+
+The following code samples outline how to pass trace context in outgoing events to some technologies in Node and Python:
+
+{{< tabs >}}
+{{% tab "Python" %}}
+
+In Python, you can use the `get_dd_trace_context` helper function to easily pass tracing context to outgoing events:
+
+```py
+import json
+import boto3
+import os
+
+from datadog_lambda.tracing import get_dd_trace_context  # Datadog tracing helper function
+
+SNS_CLIENT = boto3.client('sns')
+
+def handler(event, context):
+    SNS_CLIENT.publish(
+        TopicArn = os.getenv('SNS_ARN'),
+        Message = 'Asynchronously invoking a Lambda function with SNS.',
+        MessageAttributes = {
+            '_datadog': {
+                'DataType': 'String',
+                'StringValue': json.dumps(get_dd_trace_context()) # Includes trace context in outgoing message to SNS topic.
+            },
+        },
+    )
+```
```

Comment:
Any reason why don't we give out examples for EventBridge and Kinesis? I felt that in reality most of developers would give us or feel frustrated if there isn't an example to copy&paste. Also feels weird when we have sample extractor code for those.